### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by houdamzari**
- [x] Create model classes for all entities as shown in the diagram.
- [x] Created model methods .

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments_counter + 1)
+  end
```
Comment (AshakaE) [inline app/models/comment.rb:7]: _`Quick tip`_ - You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new post has been saved to database, then make the `update_comments_counter` method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes_counter + 1)
+  end
```
Comment (AshakaE) [inline app/models/like.rb:7]: Same here

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  def update_post_counter
+    author.update(posts_counter: author.posts_counter + 1)
+  end
```
Comment (AshakaE) [inline app/models/post.rb:8]: and here

**Review by AshakaE — APPROVED** [review]
Hi @houdamzari , 

### Highlights ✨ 

- Code base looks clean 👍 
- Pull request 👌 
- Github flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 1 above)
Comment (houdamzari) [reply app/models/comment.rb:7]: thank you for the suggestion.

**Pull request merged**